### PR TITLE
API-6945-update-token-mappings

### PR DESCRIPTION
**Pull request by jhulbert-va**
# [API-6945](https://vajira.max.gov/browse/API-6945)
In support of https://github.com/department-of-veterans-affairs/lighthouse-vulcan/pull/14 , this PR updates the deprecated tokenCsvList mappings in data-query.

### CODE REVIEW

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/condition/R4ConditionController.java [hunk 1]
```diff
@@ -213,16 +210,25 @@ private boolean tokenClinicalStatusIsSupported(TokenParameter token) {
...
-                    .map(Enum::toString)
-                    .collect(Collectors.toList()))
-        .onExplicitSystemAndExplicitCode((s, c) -> List.of(toDatamartClinicalStatusValue(c)))
-        .onAnySystemAndExplicitCode(c -> List.of(toDatamartClinicalStatusValue(c)))
+            s -> {
+              var codes =
+                  Arrays.stream(DatamartCondition.ClinicalStatus.values())
+                      .map(Enum::toString)
+                      .collect(Collectors.toList());
+              return Specifications.<ConditionEntity>selectInList("clinicalStatus", codes);
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/condition/R4ConditionController.java:222]: Would be great...
```
Specifications.<ConditionEntity>selectInList("clinicalStatus", DatamartCondition.ClinicalStatus.class);
```

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/diagnosticreport/R4DiagnosticReportController.java [hunk 2]
```diff
@@ -146,42 +146,55 @@ boolean tokenCategoryIsSupported(TokenParameter token) {
...
+                Specifications.<DiagnosticReportEntity>select(
+                    "category", CategoryCode.forFhirCategory(c).toString()))
+        .onNoSystemAndExplicitCode(
+            c ->
+                Specifications.<DiagnosticReportEntity>select(
+                    "category", CategoryCode.forFhirCategory(c).toString()))
+        .onExplicitSystemAndAnyCode(
+            s ->
+                Specifications.<DiagnosticReportEntity>selectInList(
+                    "category", Set.of(CategoryCode.CH.toString(), CategoryCode.MB.toString())))
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/diagnosticreport/R4DiagnosticReportController.java:167]: Would be great ...
```
Specifications.<DiagnosticReportEntity>selectInListOfEnums("category", CategoryCode.CH, CategoryCode.MB)
 ```                    
 Which could solve ^ the earlier "would be great" if you accept var args of Enum type. Also note you might want to leave room for other types, e.g. `selectInList("status","good","bad","cooler than a polar bear's toenails")`

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/R4ObservationController.java [hunk 3]
```diff
@@ -133,12 +136,16 @@ boolean tokenCategoryIsSupported(TokenParameter token) {
...
+  private Specification<ObservationEntity> tokenCategorySpecification(TokenParameter token) {
     return token
         .behavior()
-        .onExplicitSystemAndAnyCode(s -> List.of("laboratory", "vital-signs"))
-        .onAnySystemAndExplicitCode(List::of)
-        .onExplicitSystemAndExplicitCode((s, c) -> List.of(c))
+        .onExplicitSystemAndAnyCode(
+            s ->
+                Specifications.<ObservationEntity>selectInList(
+                    "category", List.of("laboratory", "vital-signs")))
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/R4ObservationController.java:145]: ^ hey! look at that... var args would be swell.

**Review by bschofield-va — DISMISSED** [review]
SNAPSHOT version before merge

**Review by eclendenning-va — APPROVED** [review]

**Pull request merged**